metrics-core/src/test/java/com/codahale/metrics/SlidingTimeWindowReservoirTest.java

```diff
@@ -24,6 +24,39 @@ public void storesMeasurementsWithDuplicateTicks() throws Exception {
     }
 
     @Test
+    public void storesMeasurementsWithDuplicateTicksTresholdNeg() throws Exception {
+        when(clock.getTick()).thenReturn(Long.MIN_VALUE / 256);
```

Comment:
since nanoTime can be arbitrary number (even negative per definition), let's test all possible values - this one breaks current solution on sliding-time-window-overflow branch